### PR TITLE
Simplify earlier and don't check reachability for FALSE

### PR DESCRIPTION
**Pull request by msooseth**
## Description
We can do a bit more with `simplifyProps` and we can check for `[PBool False]` during reachability check, so we don't send a `[PBool False]` to the solver

## Checklist

- [x] tested locally
- [x] added automated tests
- [ ] updated the docs
- [x] updated the changelog

### CODE REVIEW

File: src/EVM/SymExec.hs [hunk 1]
```diff
@@ -567,14 +567,14 @@ verify solvers opts preState maybepost = do
...
-          canViolate = flip filter flattened $
-            \leaf -> case Expr.simplifyProp (post preState leaf) of
-              PBool True -> False
-              _ -> True
+          -- Filter out any leaves from `flattened` that can be statically shown to be safe
+          tocheck = flip map flattened $ \leaf -> (toPropsFinal leaf, leaf)
+            where
+              toProps leaf = PNeg (post preState leaf) : assumes <> extractProps leaf
+              toPropsFinal leaf = if opts.simp then Expr.simplifyProps $ toProps leaf
+                                               else toProps leaf
```
Comment (d-xo) [inline src/EVM/SymExec.hs:575]: Can we put this in the existing `where` block? I find multiple sets of where blocks kind hard to parse

Comment (msooseth) [reply src/EVM/SymExec.hs]: Good point! I fixed it now :)

**Review by d-xo — APPROVED** [review]
lgtm :)
